README.md

```diff
@@ -9,3 +9,23 @@ Primary Goals:
   - src/main/java, src/main/test
   - relatively coarse-grained modules
 - testing: various real-ish scenarios are present in this project, and usable.
+
+
+Bazel Target Naming Conventions
+
+A lot fot hese are repeated across modules, and bazel module scoping
+provides enough namespacing information that it's redundant and arguably
+confusing to repeat the namespace in such target names. This system
+also has the benefits of being consistent horizontally and therefore
+is easily grep-able and so on.
+
+- main code for module: "lib"
+- test code for module: "test_lib"
+- executable test tasks: "tests"
+- proto: "proto"
+- java proto codegen: "proto_java"
+- grpc: "grpc"
```

Comment:
java_grpc ? if you're generating python and typescript grpc.

does typescript have grpc support yet?

Reply:
yeah but i think the point here is that you're usually not doing a bunch of heterogeneous stuff within the confines of a module. so, "grpc" in the case of python grpc would be implicit because the module would be about python and not java.

does that reasoning pass the smell test?